Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -0,0 +1,240 @@
+            m_CurrentRotation = context.ReadValue<Quaternion>();                           
+        }
+
+        protected virtual void Awake()
+        {
+           
+#if ENABLE_VR
+            if (HasStereoCamera())
+            {
+                XRDevice.DisableAutoXRCameraTracking(GetComponent<Camera>(), true);
+            }
+#endif
+        }
+
+        protected void OnEnable()
+        {
+            InputSystem.onAfterUpdate += UpdateCallback;
+            BindActions();
+        }
+
+        void OnDisable()
+        {
+            UnbindActions();
+            InputSystem.onAfterUpdate -= UpdateCallback;
+        }
+
+        protected virtual void OnDestroy()
+        {
+#if ENABLE_VR
+            if (HasStereoCamera())
+            {
+                XRDevice.DisableAutoXRCameraTracking(GetComponent<Camera>(), false);
+            }
+#endif
+        }
+
+        protected void UpdateCallback(InputUpdateType type)
+        {
+            switch(type)
+            {
+                case InputUpdateType.Fixed:
+                    FixedUpdate();
+                    break;
+                case InputUpdateType.Dynamic:
+                    OnUpdate();
+                    break;
+                case InputUpdateType.BeforeRender:
+                    OnBeforeRender();
+                    break;                
+            }
+        }
+
+        protected virtual void FixedUpdate()
+        {
+            if (m_UpdateType == UpdateType.Update ||
+                m_UpdateType == UpdateType.UpdateAndBeforeRender)
+            {
+                PerformUpdate();
+            }
+        }
+
+        protected virtual void OnUpdate()
+        {
+            if (m_UpdateType == UpdateType.Update ||
+                m_UpdateType == UpdateType.UpdateAndBeforeRender)
+            {
+                PerformUpdate();
+            }
+        }
+
+        protected virtual void OnBeforeRender()
+        {
+            if (m_UpdateType == UpdateType.BeforeRender ||
+                m_UpdateType == UpdateType.UpdateAndBeforeRender)
+            {
+                PerformUpdate();
+            }
+        }
+
+        protected virtual void SetLocalTransform(Vector3 newPosition, Quaternion newRotation)
+        {
+            if (m_TrackingType == TrackingType.RotationAndPosition ||
+                m_TrackingType == TrackingType.RotationOnly)
+            {
+                transform.localRotation = newRotation;
+            }
+
+            if (m_TrackingType == TrackingType.RotationAndPosition ||
+                m_TrackingType == TrackingType.PositionOnly)
+            {
+                transform.localPosition = newPosition;
+            }
+        }
+
+        private bool HasStereoCamera()
+        {
+            var camera = GetComponent<Camera>();
+            return camera != null && camera.stereoEnabled;
+        }
+
+        protected virtual void PerformUpdate()
+        {
+            if (!enabled)

```

Comment:
This shouldn't be necessary.  You use OnEnable/OnDisable to hook up the update loop.  That means it should always be true when hitting this.

Reply:
removed